src/Model/User.hs

```diff
@@ -733,3 +738,85 @@ pledgeStatus uid = do
     return $ case underfunded of
         [] -> AllFunded
         _ -> ExistsUnderfunded
+
+
+claimedTickets :: UserId -> Handler [(Entity Ticket, Maybe (Entity WikiTarget), Import.Value Text)]
+claimedTickets user_id = do
+
+    -- TODO: abstract out grabbing the project
+    runDB $ select $ from $ \(c `InnerJoin` t `InnerJoin` tc `LeftOuterJoin` wp `LeftOuterJoin` wt `InnerJoin` p) -> do
+        on_ $ p ^. ProjectDiscussion ==. c ^. CommentDiscussion ||. wp ?. WikiPageProject ==. just (p ^. ProjectId)
+        on_ $ wt ?. WikiTargetPage ==. wp ?. WikiPageId
+        on_ $ wp ?. WikiPageDiscussion ==. just (c ^. CommentDiscussion)
+        on_ $ tc ^. TicketClaimingTicket ==. c ^. CommentId
+        on_ $ t ^. TicketComment ==. c ^. CommentId
+
+        where_ $ tc ^. TicketClaimingUser ==. val user_id
+            &&. c ^. CommentId `notIn` (subList_select $ from $ return . (^. CommentClosingComment))
+            &&. c ^. CommentId `notIn` (subList_select $ from $ return . (^. CommentRethreadOldComment))
+
+        orderBy [ asc $ tc ^. TicketClaimingTs ]
+
+        return (t, wt, p ^. ProjectHandle)
+
+    -- XXX: There are two known issues with this query:
+    -- 1. If a watched comment is a ticket and the nth child, the
+    -- query will return the same ticket n times.
+    -- 2. If there are n watched comments in the same thread, each
+    -- child ticket in the thread will be returned n times.
+    -- 'select . distinct' just hides these problems from the user's
+    -- eyes.
+
+countClaimedTickets :: UserId -> Handler Int
+countClaimedTickets user_id = do
+
+    claimed_tickets <- claimedTickets user_id
+    return $ length claimed_tickets
+
+countWatchedTickets :: UserId -> Handler Int
+countWatchedTickets user_id = do
+
+    watched_tickets <- watchedTickets user_id
+    return $ length watched_tickets
+
+countTickets :: UserId -> Handler Int
+countTickets user_id = do
+
+    claimed_tickets <- claimedTickets user_id
+    watched_tickets <- watchedTickets user_id
+    return $ (length watched_tickets) + (length claimed_tickets)
```

Comment:
Ideally I'd do even the addition on the SQL side, because I'm like that, but should at *least* call the count functions we made above and sum those instead of re-implementing them